src/generic/grideditors.cpp

```diff
@@ -703,6 +707,40 @@ void wxGridCellNumberEditor::Create(wxWindow* parent,
     }
 }
 
+void wxGridCellNumberEditor::SetSize(const wxRect& r)
+{
+#if wxUSE_SPINCTRL
+    if ( HasRange() )
+    {
+        wxASSERT_MSG(m_control, "The wxSpinCtrl must be created first!");
+
+        const wxSize bestSize = Spin()->GetBestSize();
+        const int width = wxMax(r.GetWidth(), bestSize.GetWidth());
+        const int height = bestSize.GetHeight() > 0 ?
+                           bestSize.GetHeight() :
+                           r.GetHeight();
+
+        wxRect rect(r.GetLeft(), r.GetTop(), width, bestSize.GetHeight());
+
+        // If edit not the top most cell then align center.
```

Comment:
```suggestion
        // If possible, i.e. if we're not editing the topmost or leftmost cell, center the control rectangle in the cell.
```

BTW, it would be great to rename `r` and `rect` to `rectCell` and `rectSpin` or something like this to make it immediately clear which one is which -- which is not at all the case now.